USAGE.md

```diff
@@ -3,34 +3,80 @@
 ## 1. Vault Enterprise License
 Retrieve a [Vault Enterprise trial license](https://www.hashicorp.com/products/vault/trial).
 
-Export the license in your shell environment:
-```yaml
-export VAULT_LICENSE=02MV4UU..
-```
-
-## 2. Start Containers
+## 2. Setup environment
+Export environment variables in a shell for
+
+* the Vault license
+  ```vash
+  export VAULT_LICENSE=02MV4UU..
+  ```
+* the runtime you're going to use
+  ```bash
+  export CONTAINER_RUNTIME=${CONTAINER_RUNTIME:-docker}
+  ```
+  *The default is `docker`. If podman is going to be used, change this to `podman`.*
+* the internal domain containers can be addressed with
+  ```bash
+  if [[ -z ${CONTAINER_RUNTIME##docker} ]];then export CONTAINER_DOMAIN=docker.internal; else export CONTAINER_DOMAIN=dns.podman; fi
+  ```
+  *uses `docker.internal` if docker is going to be used and `dns.podman` if podman. This does not need modification unless another domain name is required.*
+* the realm domain for the freeipa server
+  ```bash
+  export IPA_REALM=${CONTAINER_DOMAIN^^}
+  ```
+
+Or create a `.env` file for `(docker|podman)-compose`
+```bash
+# Example for podman
+CONTAINER_RUNTIME=podman
+VAULT_LICENSE=02MV4UU..
+echo CONTAINER_RUNTIME=${CONTAINER_RUNTIME:-docker} > .env
+if [[ -z ${CONTAINER_RUNTIME##docker} ]];then echo CONTAINER_DOMAIN=docker.internal; else echo CONTAINER_DOMAIN=dns.podman; fi >> .env
+echo VAULT_LICENSE=${VAULT_LICENSE} >> .env
+echo IPA_REALM=${CONTAINER_DOMAIN^^} >> .env
+```
+
+## 3. Start Containers
 
 Start FreeIPA, Keycloak, Vault and K3s:
 ```bash
-docker-compose up -d
+sudo ${CONTAINER_RUNTIME}-compose up -d
 ```
 
-## 3. Setup Resolution of Domain Names
+## 4. Setup Resolution of Domain Names
+
+You can either add a static record for every container to your `/etc/hosts` file or configure systemd-resolved service with `resolvectl` to use the container environment internal DNS (recommended).
+
+### Option 1: Configuration using container internal dns (recommended)
+First, find out which interface is used
+```bash
+CONTAINER_RUNTIME_IF=$(sudo ${CONTAINER_RUNTIME} network inspect vault-playground_vault-playground | jq -j '.[].network_interface')
```

Comment:
- Is `sudo` required here?
- The JSON output of docker/podman for network inspection are very different:
  * https://docs.podman.io/en/latest/markdown/podman-network-inspect.1.html#example
  * https://docs.docker.com/engine/reference/commandline/network_inspect
- My output for `docker network inspect` does not show the key `network_interface`. I think that output is `$CONTAINER_RUNTIME` specific:
```bash
${CONTAINER_RUNTIME} network inspect vault-playground_vault-playground
[
    {
        "Name": "vault-playground_vault-playground",
        "Id": "e6a39916fba2ea3ddbc7daee454f15c616aa0bd7dda36eed09a096517d7d1fff",
        "Created": "2022-07-29T08:45:05.881540185+02:00",
        "Scope": "local",
        "Driver": "bridge",
        "EnableIPv6": false,
        "IPAM": {
            "Driver": "default",
            "Options": null,
            "Config": [
                {
                    "Subnet": "172.18.0.0/16",
                    "Gateway": "172.18.0.1"
                }
            ]
        },
        "Internal": false,
        "Attachable": true,
        "Ingress": false,
        "ConfigFrom": {
            "Network": ""
        },
        "ConfigOnly": false,
        "Containers": {
            "28e05d32155df21d7c87b93384786b3622c2548917461d6ac48560fbe7ee81b8": {
                "Name": "grafana",
                "EndpointID": "3ffd779d3373ca9bbacdc3cc8ba03e0fafafc0109a5d062d7bcf7643215ee5b1",
                "MacAddress": "<redacted>",
                "IPv4Address": "172.18.0.5/16",
                "IPv6Address": ""
            },
            ... <redacted> ...
            "e4c0874850083be1401519cf049cb10c5c7075bcb0bb12a653266b879f1e6d45": {
                "Name": "vault",
                "EndpointID": "829b819ec3281abe3b3f59de9f07c0085d3c87c1357a2435060ee82f0baa19e5",
                "MacAddress": "<redacted>",
                "IPv4Address": "172.18.0.4/16",
                "IPv6Address": ""
            }
        },
        "Options": {},
        "Labels": {
            "com.docker.compose.network": "vault-playground",
            "com.docker.compose.project": "vault-playground",
            "com.docker.compose.version": "1.29.2"
        }
    }
]
```

Reply:
Looks like a script should be created to manage those differences.

Reply:
okay, thanks! then we'll have a look to find another way to find the exact interface created

Reply:
https://github.com/adfinis-sygroup/vault-playground/commit/db3dbe31b1b854d80acae33832cd1fb9659fde50